src/transaction.js

```diff
@@ -435,11 +450,28 @@ const transaction = {
 
     // Update incompleteTxData.
     Object.assign(incompleteTxData, newData);
+  },
 
+  /**
+   * Update weight from tx data (if not set yet)
+   *
+   * @param {Object} data Object with complete tx data
+   * {
+   *  'inputs': [{'tx_id', 'index'}],
+   *  'outputs': ['address', 'value', 'timelock'],
+   *  'nonce': 0,
+   *  'version': 1,
+   *  'timestamp': 123,
+   * }
+   *
+   * @memberof Transaction
+   * @inner
+   */
+  setWeight(data) {
     // Calculate tx weight if needed.
-    if (incompleteTxData.weight == 0) {
-      let minimumWeight = this.calculateTxWeight(incompleteTxData);
-      incompleteTxData['weight'] = minimumWeight;
+    if (data.weight === 0) {
```

Comment:
Why would we do this `if`? Maybe add a `force` parameter? Or create a `setWeightIfNeeded`?

Reply:
The `if` is in case the user does not want to set the weight, we calculate it, otherwise we use what the user set.

Maybe could check if `if ('weight' in data)`. What do you think?

Reply:
I think that if the user wants to calculate the weight, they just skip calling this method. Or they could call another method called `setWeightIfNeeded`. But I don't like skipping the weight calculation when I call `setWeight`. I think this might be error prone. What do you think?

Reply:
I agree. But currently we don't have any case where we always want to set weight. So I will just update the name of this method to `setWeightIfNeeded` to avoid confusion. Is it ok?

Reply:
I think it is. Thanks!